No longer builds on latest ARCH linux

Installing/building the native extension is no longer possible on latest Arch-Linux.

Kernel 6.10.9
GCC 14.2.1


The compilation fails with following error (only presenting the last lines of the output):

```
sigar_util.c:652:19: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  652 |     CPU_MODEL_ENT("Duron"),
      |                   ^~~~~~~
sigar_util.c:627:7: note: in definition of macro ‘CPU_MODEL_ENT_R’
  627 |     { s, sizeof(s)-1, r, sizeof(r) }
      |       ^
sigar_util.c:652:5: note: in expansion of macro ‘CPU_MODEL_ENT’
  652 |     CPU_MODEL_ENT("Duron"),
      |     ^~~~~~~~~~~~~
sigar_util.c:652:19: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  652 |     CPU_MODEL_ENT("Duron"),
      |                   ^~~~~~~
sigar_util.c:627:23: note: in definition of macro ‘CPU_MODEL_ENT_R’
  627 |     { s, sizeof(s)-1, r, sizeof(r) }
      |                       ^
sigar_util.c:652:5: note: in expansion of macro ‘CPU_MODEL_ENT’
  652 |     CPU_MODEL_ENT("Duron"),
      |     ^~~~~~~~~~~~~
sigar_util.c:653:21: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  653 |     CPU_MODEL_ENT_R("K6(tm)-III", "K6 III"),
      |                     ^~~~~~~~~~~~
sigar_util.c:627:7: note: in definition of macro ‘CPU_MODEL_ENT_R’
  627 |     { s, sizeof(s)-1, r, sizeof(r) }
      |       ^
sigar_util.c:653:35: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  653 |     CPU_MODEL_ENT_R("K6(tm)-III", "K6 III"),
      |                                   ^~~~~~~~
sigar_util.c:627:23: note: in definition of macro ‘CPU_MODEL_ENT_R’
  627 |     { s, sizeof(s)-1, r, sizeof(r) }
      |                       ^
sigar_util.c:654:21: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  654 |     CPU_MODEL_ENT_R("K6(tm) 3D+", "K6 3D+"),
      |                     ^~~~~~~~~~~~
sigar_util.c:627:7: note: in definition of macro ‘CPU_MODEL_ENT_R’
  627 |     { s, sizeof(s)-1, r, sizeof(r) }
      |       ^
sigar_util.c:654:35: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  654 |     CPU_MODEL_ENT_R("K6(tm) 3D+", "K6 3D+"),
      |                                   ^~~~~~~~
sigar_util.c:627:23: note: in definition of macro ‘CPU_MODEL_ENT_R’
  627 |     { s, sizeof(s)-1, r, sizeof(r) }
      |                       ^
sigar_util.c: In function ‘sigar_log_printf’:
sigar_util.c:919:5: warning: function ‘sigar_log_printf’ might be a candidate for ‘gnu_printf’ format attribute [-Wsuggest-attribute=format]
  919 |     vsnprintf(buffer, sizeof(buffer), format, args);
      |     ^~~~~~~~~
sigar_util.c: In function ‘sigar_time_now_millis’:
sigar_util.c:980:5: error: implicit declaration of function ‘gettimeofday’ [-Wimplicit-function-declaration]
  980 |     gettimeofday(&tv, NULL);
      |     ^~~~~~~~~~~~
At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may have been intended to silence earlier diagnostics
make: *** [Makefile:248: sigar_util.o] Error 1

make failed, exit code 2

```

--- a/Rakefile
+++ b/Rakefile
@@ -51,15 +51,15 @@ end
 desc 'Build sigar extension'
 task :build do
   in_ext()
-  unless File.exists? "Makefile"
+  unless File.exist? "Makefile"
     unless system("ruby extconf.rb")
       STDERR.puts "Failed to configure"
-      break
+      next
     end
   end
   unless system(MAKE)
     STDERR.puts 'Failed to ' + MAKE
-    break
+    next
   end
 end
 
@@ -75,13 +75,13 @@ end
 desc 'Clean sigar extension'
 task :clean do
   in_ext()
-  system(MAKE + ' clean') if File.exists? "Makefile"
+  system(MAKE + ' clean') if File.exist? "Makefile"
 end
 
 desc 'Dist Clean sigar extension'
 task :distclean do
   in_ext()
-  system(MAKE + ' distclean') if File.exists? "Makefile"
+  system(MAKE + ' distclean') if File.exist? "Makefile"
 end
 
 desc 'Run sigar examples (test)'

--- a/src/sigar_util.c
+++ b/src/sigar_util.c
@@ -30,6 +30,7 @@
 
 #include <dirent.h>
 #include <sys/stat.h>
+#include <sys/time.h>
 
 SIGAR_INLINE char *sigar_uitoa(char *buf, unsigned int n, int *len)
 {

